Source/WebCore/platform/graphics/BitmapImageSource.cpp

```diff
@@ -607,6 +602,11 @@ Expected<Ref<NativeImage>, DecodingStatus> BitmapImageSource::nativeImageAtIndex
 Expected<Ref<NativeImage>, DecodingStatus> BitmapImageSource::currentNativeImageForDrawing(SubsamplingLevel subsamplingLevel, const DecodingOptions& options)
 {
     startAnimation(subsamplingLevel, options);
+
+    // The first frame has to be decoded synchronously for the first time.
+    if (options.decodingMode() == DecodingMode::Asynchronous && isAnimated() && !hasEverAnimated())
+        return nativeImageAtIndexForDrawing(currentFrameIndex(), subsamplingLevel, { DecodingMode::Synchronous, options.sizeForDrawing() });
```

Comment:
Rather than repeat the call to `nativeImageAtIndexForDrawing()` inside the clause here, it would be cleaner to just adjust `options` and have a single call to `nativeImageAtIndexForDrawing()`

Reply:
Done.